File: 205-isomorphic-strings/isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {

        if(s.length()!=t.length()) return false;

        Map<Character,Character> s2T = new HashMap<>();
        Map<Character,Character> t2S = new HashMap<>();

        for(int i=0;i<s.length();i++){

            char sChar = s.charAt(i);
            char tChar = t.charAt(i);

            if (!s2T.containsKey(sChar) && !t2S.containsKey(tChar)) {
                s2T.put(sChar, tChar);
                t2S.put(tChar, sChar);
            } 

            

            if (s2T.containsKey(sChar) != t2S.containsKey(tChar)) return false;

            else if(s2T.get(sChar)!=tChar || t2S.get(tChar)!=sChar) return false;
        }
        System.out.print(s2T);
        System.out.print(t2S);

        return true;

    }
}

File: 205-isomorphic-strings/isomorphic-strings.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isIsomorphic = function(s, t) {

    if(s.length!==t.length) return false;

    let s2T = new Map();
    let t2S = new Map();

    for(let i=0;i<s.length;i++) {

        if(!s2T.has(s[i]) && !t2S.has(t[i])){
            t2S.set(t[i],s[i]);
            s2T.set(s[i],t[i]);
        }
        else if(t2S.get(t[i])!==s[i] || s2T.get(s[i])!==t[i]) return false;
    }

    return true;


    
};